axelor-account/src/main/java/com/axelor/apps/account/service/move/MoveLineInvoiceTermServiceImpl.java

```diff
@@ -405,4 +410,26 @@ public void setDueDateFromInvoiceTerms(MoveLine moveLine) {
     moveLine.setDueDate(
         invoiceTermService.getDueDate(moveLine.getInvoiceTermList(), moveLine.getDueDate()));
   }
+
+  protected void adjustLastInvoiceTerm(List<InvoiceTerm> invoiceTermList, BigDecimal totalAmount) {
+    BigDecimal sumOfInvoiceTerm =
+        invoiceTermList.stream()
+            .map(InvoiceTerm::getAmount)
+            .reduce(BigDecimal::add)
+            .orElse(BigDecimal.ZERO);
+    if (totalAmount.compareTo(sumOfInvoiceTerm) != 0) {
+      InvoiceTerm lastElement =
+          invoiceTermList.stream().reduce((first, second) -> second).orElse(null);
+      if (lastElement == null) {
+        return;
+      }
+
+      BigDecimal difference = totalAmount.subtract(sumOfInvoiceTerm);
+      BigDecimal amount = lastElement.getAmount().add(difference);
+      lastElement.setAmount(amount);
+      lastElement.setAmountRemaining(amount);
```

Comment:
Is there always amount = amountRemaining when this method is called ? What about partial payments ?